packages/flame/lib/src/components/sprite_animation_component.dart

```diff
@@ -23,42 +24,66 @@ class SpriteAnimationComponent extends PositionComponent with HasPaint {
 
   /// Creates a component with an empty animation which can be set later
   SpriteAnimationComponent({
+    this.animation,
+    bool? removeOnFinish,
+    bool? playing,
+    Paint? paint,
     Vector2? position,
     Vector2? size,
+    Vector2? scale,
+    double? angle,
+    Anchor? anchor,
     int? priority,
-    this.animation,
-    Paint? paint,
-    this.removeOnFinish = false,
-    this.playing = true,
-  }) : super(position: position, size: size, priority: priority) {
+  })  : removeOnFinish = removeOnFinish ?? false,
```

Comment:
won't this generate some shadowing?

Reply:
It doesn't, it is a strange feature that dart has. Only variables that are in the class scope can be assigned here, but you can still use variables from the constructor.

Reply:
wouldn't it be simpler to say `this.removeOnFinish = false`?

Reply:
Since `null` will be passed in from the named constructor `SpriteAnimationComponent.fromFrameData` and we don't want to set the defaults in two places the `this.` notation can't be used.